mockgcp/mockspanner/admin/database.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockspanner
+
+import (
+	"context"
+	"strings"
+
+	"cloud.google.com/go/longrunning/autogen/longrunningpb"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/common/projects"
+	pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/spanner/admin/database/v1"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"k8s.io/klog/v2"
+)
+
+var _ pb.DatabaseAdminServer = &SpannerDatabaseV1{}
+
+type SpannerDatabaseV1 struct {
+	*MockService
+	pb.UnimplementedDatabaseAdminServer
+}
+
+type spannerDatabaseName struct {
+	Project      *projects.ProjectData
+	InstanceName string
+	DatabaseName string
+}
+
+func (s *SpannerDatabaseV1) GetDatabase(ctx context.Context, req *pb.GetDatabaseRequest) (*pb.Database, error) {
+	name, err := s.parseDatabaseName(req.GetName())
+	if err != nil {
+		return nil, err
+	}
+	fqn := name.String()
+
+	obj := &pb.Database{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+func (s *SpannerDatabaseV1) CreateDatabase(ctx context.Context, req *pb.CreateDatabaseRequest) (*longrunningpb.Operation, error) {
+	databaseName := databaseNameFromCreateStatement(req.GetCreateStatement())
+	fqn := req.GetParent() + "/databases/" + databaseName
+
+	obj := &pb.Database{}
+	obj.Name = fqn
+
+	now := timestamppb.Now()
+	obj.CreateTime = now
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	return s.operations.NewLRO(ctx)
```

Comment:
I think we would want to use StartLRO if possible which better reflects the actual service behavior. (Feel free to make the improvement in follow up PRs.)

Reply:
Ok, I will look into this. Thanks 👍